python/test/hail/methods/test_statgen.py

```diff
@@ -1080,3 +1080,15 @@ def test_window_by_locus(self):
         self.assertTrue(entries.all(hl.all(lambda x: x.e_col_idx == entries.col_idx, entries.prev_entries)))
         self.assertTrue(entries.all(hl.all(lambda x: entries.row_idx - 1 - x[0] == x[1].e_row_idx,
                                            hl.zip_with_index(entries.prev_entries))))
+
+    def test_warn_if_no_intercept(self):
+        mt = hl.balding_nichols_model(1, 1, 1).add_row_index().add_col_index()
+        intercept = hl.float64(1.0)
+
+        for covariates in [[],
+                           [mt.row_idx],
+                           [mt.col_idx],
+                           [mt.GT.n_alt_alleles()],
+                           [mt.row_idx, mt.col_idx, mt.GT.n_alt_alleles()]]:
+            self.assertTrue(hl.methods.statgen._warn_if_no_intercept('', covariates))
+            self.assertFalse('', [intercept] + covariates)
```

Comment:
you didn't call `_warn_if_no_intercept` in your assertFalse test.

Reply:
oops, yes! (and i made sure the test fails if I don't include intercept in the second line)